src/Components/Server/src/Circuits/CircuitHost.cs

```diff
@@ -258,53 +258,39 @@ public async Task OnConnectionUpAsync(CancellationToken cancellationToken)
         {
             Log.ConnectionUp(_logger, Circuit.Id, Client.ConnectionId);
 
-            try
-            {
-                await HandlerLock.WaitAsync(cancellationToken);
+            Renderer.Dispatcher.AssertAccess();
```

Comment:
We should do this in `OnCircuitOpenedAsync` also

Reply:
Done